pkg/p2p/libp2p/connections_test.go

```diff
@@ -598,6 +599,43 @@ func TestBlocklisting(t *testing.T) {
 	expectPeers(t, s2)
 }
 
+func TestBlocklistedPeers(t *testing.T) {
+	t.Parallel()
+	s1, overlay1 := newService(t, 1, libp2pServiceOpts{libp2pOpts: libp2p.Options{
+		FullNode: true,
+	}})
+	s2, _ := newService(t, 1, libp2pServiceOpts{})
+	addr1 := serviceUnderlayAddress(t, s1)
+	_, err := s2.Connect(context.Background(), addr1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err = s2.Blocklist(overlay1, 0, testBlocklistMsg); err != nil {
+		t.Fatal(err)
+	}
+	blocklistedPeers, err := s2.BlocklistedPeers()
+	if err != nil {
+		t.Fatal("could not get blocklisted peers", err)
+	}
+	if len(blocklistedPeers) != 1 {
+		t.Fatalf("expected %d blocklisted peers, got %d", 1, len(blocklistedPeers))
+	}
+
+	expected := p2p.BlockListedPeer{
+		Peer: p2p.Peer{
+			Address:  overlay1,
+			FullNode: true,
+		},
+		Reason:   testBlocklistMsg,
+		Duration: 0,
+	}
+
+	if !reflect.DeepEqual(expected, blocklistedPeers[0]) {
+		t.Fatal("expected blocklisted peer", expected, "got", blocklistedPeers[0])
+	}
```

Comment:
nit: could `expected` variable be of the same type as `blocklistedPeers`, that is `[]p2p.BlockListedPeer`? this way first check `if len(blocklistedPeers) != 1 ` would be unnecessary?